oml/miners/inbatch_nhard_tri.py

```diff
@@ -0,0 +1,140 @@
+from typing import Any, List, Optional, Tuple, Union
+
+import numpy as np
+import torch
+from torch import Tensor
+
+from oml.interfaces.miners import ITripletsMinerInBatch, TTripletsIds
+from oml.utils.misc_torch import pairwise_dist
+
+
+class NHardTripletsMiner(ITripletsMinerInBatch):
+    """
+    This miner selects top hard triplets based on distances between features:
+
+    - hard `positive` samples have large distance to the anchor sample
+
+    - hard `negative` samples have small distance to the anchor sample
+
+    Toward the end of the training, annotation errors can affect final metric. If you are not sure about the quality of
+    your dataset, you can use range instead of integer value for paramters and exclude combinations with the largest
+    distances. For example instead picking `top5` positive examples, you can use `top2 - top5` examples.
```

Comment:
`top5` -> 5 
`top2 - top5` -> you can use examples from the 2nd hardest to the 5th one.